src/progress.rs

```diff
@@ -48,9 +54,81 @@ impl Default for ProgressState {
 }
 
 #[derive(Clone, Debug, Default)]
-struct Configuration {
-    voters: FxHashSet<u64>,
-    learners: FxHashSet<u64>,
+pub struct Configuration {
+    pub voters: FxHashSet<u64>,
+    pub learners: FxHashSet<u64>,
+}
+
+impl Configuration {
+    pub fn new(
+        voters: impl IntoIterator<Item = u64>,
+        learners: impl IntoIterator<Item = u64>,
+    ) -> Self {
+        Self {
+            voters: voters.into_iter().collect(),
+            learners: learners.into_iter().collect(),
+        }
+    }
+}
+
+impl<'a> From<&'a ConfState> for Configuration {
+    fn from(conf_state: &'a ConfState) -> Self {
+        Self {
+            voters: conf_state.get_nodes().iter().cloned().collect(),
+            learners: conf_state.get_learners().iter().cloned().collect(),
+        }
+    }
+}
+
+impl Into<ConfState> for Configuration {
+    fn into(self) -> ConfState {
+        let mut state = ConfState::new();
+        state.set_nodes(self.voters.iter().cloned().collect());
+        state.set_learners(self.learners.iter().cloned().collect());
+        state
+    }
+}
+
+impl Configuration {
+    fn with_capacity(voters: usize, learners: usize) -> Self {
+        Self {
+            voters: HashSet::with_capacity_and_hasher(voters, FxBuildHasher::default()),
+            learners: HashSet::with_capacity_and_hasher(learners, FxBuildHasher::default()),
+        }
+    }
+
+    // Validates that the configuration not problematic.
+    //
+    // Namely:
+    // * There can be no overlap of voters and learners.
+    // * There must be at least one voter.
+    pub fn valid(&self) -> Result<(), Error> {
```

Comment:
How about `validate`? It's a verb.

Reply:
I think `config.valid()?;` reads like a question, a bit nicer than `config.validate()?`, what do you think?

Reply:
Got it! :)